src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -689,6 +697,64 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
     }
   }
 
+  private runFinalPreSubmitUpdate() {
+    // Even if next guard fails, we want to prevent attempting to pay.
+    this.runningFinalPreSubmitUpdate = true;
+
+    let requiredDonationProperties = [this.donation, this.campaign, this.campaign.charity.id, this.psp];
+    // Return if any of the required properties are missing.
```

Comment:
I'm not seeing a return statement here, am I missing something?

Reply:
Nope think this is from before I moved to using `forEach()`, fixed in b37a54e471683882629b389d9c9964de067f68f3